trace/Span.md

```diff
@@ -0,0 +1,32 @@
+# Span API
+
+This class represents a Span. A span contains a SpanContext and  allows users to record 
+tracing events based on the data model defined [here](https://github.com/census-instrumentation/opencensus-proto/blob/master/trace/trace.proto).
+All Span implementations MUST not allow users to read the recorded data, 
```

Comment:
This is an interesting requirement. Why is this important?

Reply:
Probably this is about the span api, as there've been some things in the past where there's readback race conditions which aren't worth explaining. Brave used to have synchronized readback, we removed it, and will likely only support this as a pre-report hook. Not sure bogdan's rationale though!

Reply:
So I would like the OpenCensus libraries to have the same "look and feel". I think it is hard to get it right in all the languages (if we really need it we can discuss and add it later but I need to think more about what is really needed). Removing an API is much harder than adding a new one.

The rational behind this is the following:
* I would like to achieve the same "look and feel" for all the languages.
* If you allow  any getter then that means the data needs to be stored in the Span object all the time and they cannot be streamed directly to the repository for example.
* If you have to store all the data because you want to provide getters then you for refcounting (in languages like C++ where no GC).
* If you force refcounting you force heap-allocation, which in some C++ systems may be expensive.

I think for the moment adding getters in some languages limits the implementation design and I would prefer to not have it until we have a strong argument to support that.